compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2092,9 +2112,21 @@ public void exitSimpleLiteral(BallerinaParser.SimpleLiteralContext ctx) {
         } else if (ctx.NullLiteral() != null || ctx.emptyTupleLiteral() != null) {
             this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.NIL, null);
         } else if (ctx.blobLiteral() != null) {
-            byte[] blobValue = getBlobLiteral(ctx.blobLiteral());
-            this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.BLOB, blobValue);
+            this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.ARRAY, ctx.blobLiteral().getText());
+        }
+    }
+
+    private boolean isByteLiteral(BallerinaParser.SimpleLiteralContext ctx, Long longObject) {
+        if (!(ctx.parent.parent instanceof BallerinaParser.VariableDefinitionStatementContext)) {
```

Comment:
This case won't work in some cases. Like foo(5), where foo takes a byte argument. I think we don't need this validation, as we do the same at type checker.